From: Kevin Nelson <github.com/knelsonmeister>
Date: Wed, 12 Mar 2025 20:12:34 +0500
Subject: trixie-support

Index: systemd/src/shared/bus-unit-util.c
===================================================================
--- systemd.orig/src/shared/bus-unit-util.c
+++ systemd/src/shared/bus-unit-util.c
@@ -39,6 +39,7 @@
 #include "percent-util.h"
 #include "process-util.h"
 #include "rlimit-util.h"
+#include "sd-daemon.h"
 #include "seccomp-util.h"
 #include "securebits-util.h"
 #include "signal-util.h"
@@ -2906,10 +2907,12 @@ int bus_append_scope_pidref(sd_bus_messa
         if (!pidref_is_set(pidref))
                 return -ESRCH;

-        if (pidref->fd >= 0 && allow_pidfd)
-                return sd_bus_message_append(
-                                m, "(sv)",
-                                "PIDFDs", "ah", 1, pidref->fd);
+        if (sd_booted()) {
+                if (pidref->fd >= 0 && allow_pidfd)
+                        return sd_bus_message_append(
+                                        m, "(sv)",
+                                        "PIDFDs", "ah", 1, pidref->fd);
+        }

         return sd_bus_message_append(
                         m, "(sv)",
Index: systemd/src/login/logind-session.c
===================================================================
--- systemd.orig/src/login/logind-session.c
+++ systemd/src/login/logind-session.c
@@ -1338,8 +1338,10 @@ bool session_may_gc(Session *s, bool dro
                 return true;
         if (r < 0)
                 log_debug_errno(r, "Unable to determine if leader PID " PID_FMT " is still alive, assuming not: %m", s->leader.pid);
-        if (r > 0)
-                return false;
+        if (sd_booted()) {
+                if (r > 0)
+                        return false;
+        }

         if (s->fifo_fd >= 0 && pipe_eof(s->fifo_fd) <= 0)
                 return false;
Index: systemd/src/login/logind-user.c
===================================================================
--- systemd.orig/src/login/logind-user.c
+++ systemd/src/login/logind-user.c
@@ -874,18 +874,34 @@ bool user_may_gc(User *u, bool drop_not_

         /* Check if our job is still pending */
         const char *j;
-        FOREACH_ARGUMENT(j, u->runtime_dir_job, u->service_manager_job) {
-                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        if (sd_booted()) {
+                FOREACH_ARGUMENT(j, u->runtime_dir_job, u->service_manager_job) {
+                        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;

-                if (!j)
-                        continue;
+                        if (!j)
+                                continue;

-                r = manager_job_is_active(u->manager, j, &error);
-                if (r < 0)
-                        log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
-                                        j, bus_error_message(&error, r));
-                if (r != 0)
-                        return false;
+                        r = manager_job_is_active(u->manager, j, &error);
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
+                                                j, bus_error_message(&error, r));
+                        if (r != 0)
+                                return false;
+                }
+        } else {
+                FOREACH_ARGUMENT(j, u->service_manager_job) {
+                        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+
+                        if (!j)
+                                continue;
+
+                        r = manager_job_is_active(u->manager, j, &error);
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
+                                                j, bus_error_message(&error, r));
+                        if (r != 0)
+                                return false;
+                }
         }

         /* Note that we don't care if the three units we manage for each user object are up or not, as we are
Index: systemd/src/test/test-mempress.c
===================================================================
--- systemd.orig/src/test/test-mempress.c
+++ systemd/src/test/test-mempress.c
@@ -7,6 +7,7 @@

 #include <sd-bus.h>
 #include <sd-event.h>
+#include <sd-daemon.h>

 #include "bus-locator.h"
 #include "bus-wait-for-jobs.h"
@@ -264,7 +265,8 @@ TEST(real_pressure) {
         assert_se(uo = unit_dbus_path_from_name(scope));

         uint64_t mcurrent = UINT64_MAX;
-        assert_se(sd_bus_get_property_trivial(bus, "org.freedesktop.systemd1", uo, "org.freedesktop.systemd1.Scope", "MemoryCurrent", &error, 't', &mcurrent) >= 0);
+        if (sd_booted())
+                assert_se(sd_bus_get_property_trivial(bus, "org.freedesktop.systemd1", uo, "org.freedesktop.systemd1.Scope", "MemoryCurrent", &error, 't', &mcurrent) >= 0);

         printf("current: %" PRIu64 "\n", mcurrent);
         if (mcurrent == UINT64_MAX) {
Index: systemd/src/fuzz/fuzz-compress.c
===================================================================
--- systemd.orig/src/fuzz/fuzz-compress.c
+++ systemd/src/fuzz/fuzz-compress.c
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */

 #include <errno.h>
+#include <sd-daemon.h>

 #include "alloc-util.h"
 #include "compress.h"
@@ -22,7 +23,10 @@ int LLVMFuzzerTestOneInput(const uint8_t
         _cleanup_free_ void *buf = NULL, *buf2 = NULL;
         int r;

+        if (!sd_booted())
+                return EXIT_TEST_SKIP;
+
         if (size < offsetof(header, data) + 1)
                 return 0;

         const header *h = (struct header*) data;
