From: Kevin Nelson <github.com/knelsonmeister>
Date: Wed, 12 Mar 2025 20:12:34 +0500
Subject: trixie-support

Index: systemd/debian/control
===================================================================
--- systemd.orig/debian/control
+++ systemd/debian/control
@@ -111,7 +111,6 @@ Depends: ${shlibs:Depends},
          mount,
 Conflicts: consolekit,
            libpam-ck-connector,
-           systemd-shim,
            opensysusers,
 Breaks: less (<< 563),
         sicherboot (<< 0.1.6),
@@ -246,7 +245,7 @@ Depends: ${shlibs:Depends},
          systemd (= ${binary:Version}),
          libpam-runtime,
          default-dbus-system-bus | dbus-system-bus,
-         systemd-sysv
+         systemd-shim (>= 10-5~) | systemd-sysv
 Recommends: dbus-user-session,
             ${dlopen:Recommends},
 Suggests: ${dlopen:Suggests},
Index: systemd/src/shared/bus-unit-util.c
===================================================================
--- systemd.orig/src/shared/bus-unit-util.c
+++ systemd/src/shared/bus-unit-util.c
@@ -39,6 +39,7 @@
 #include "percent-util.h"
 #include "process-util.h"
 #include "rlimit-util.h"
+#include "sd-daemon.h"
 #include "seccomp-util.h"
 #include "securebits-util.h"
 #include "signal-util.h"
@@ -2828,10 +2829,12 @@ int bus_append_scope_pidref(sd_bus_messa
         if (!pidref_is_set(pidref))
                 return -ESRCH;

-        if (pidref->fd >= 0 && allow_pidfd)
-                return sd_bus_message_append(
-                                m, "(sv)",
-                                "PIDFDs", "ah", 1, pidref->fd);
+        if (sd_booted()) {
+                if (pidref->fd >= 0 && allow_pidfd)
+                        return sd_bus_message_append(
+                                        m, "(sv)",
+                                        "PIDFDs", "ah", 1, pidref->fd);
+        }

         return sd_bus_message_append(
                         m, "(sv)",
Index: systemd/src/login/logind-session.c
===================================================================
--- systemd.orig/src/login/logind-session.c
+++ systemd/src/login/logind-session.c
@@ -1326,8 +1326,10 @@ bool session_may_gc(Session *s, bool dro
                 return true;
         if (r < 0)
                 log_debug_errno(r, "Unable to determine if leader PID " PID_FMT " is still alive, assuming not: %m", s->leader.pid);
-        if (r > 0)
-                return false;
+        if (sd_booted()) {
+                if (r > 0)
+                        return false;
+        }

         if (s->fifo_fd >= 0 && pipe_eof(s->fifo_fd) <= 0)
                 return false;
Index: systemd/src/login/logind-user.c
===================================================================
--- systemd.orig/src/login/logind-user.c
+++ systemd/src/login/logind-user.c
@@ -27,6 +27,7 @@
 #include "path-util.h"
 #include "percent-util.h"
 #include "rm-rf.h"
+#include "sd-daemon.h"
 #include "serialize.h"
 #include "special.h"
 #include "stdio-util.h"
@@ -782,18 +783,34 @@ bool user_may_gc(User *u, bool drop_not_

         /* Check if our job is still pending */
         const char *j;
-        FOREACH_ARGUMENT(j, u->runtime_dir_job, u->service_manager_job) {
-                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        if (sd_booted()) {
+                FOREACH_ARGUMENT(j, u->runtime_dir_job, u->service_manager_job) {
+                        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;

-                if (!j)
-                        continue;
+                        if (!j)
+                                continue;

-                r = manager_job_is_active(u->manager, j, &error);
-                if (r < 0)
-                        log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
-                                        j, bus_error_message(&error, r));
-                if (r != 0)
-                        return false;
+                        r = manager_job_is_active(u->manager, j, &error);
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
+                                                j, bus_error_message(&error, r));
+                        if (r != 0)
+                                return false;
+                }
+        } else {
+                FOREACH_ARGUMENT(j, u->service_manager_job) {
+                        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+
+                        if (!j)
+                                continue;
+
+                        r = manager_job_is_active(u->manager, j, &error);
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to determine whether job '%s' is pending, ignoring: %s",
+                                                j, bus_error_message(&error, r));
+                        if (r != 0)
+                                return false;
+                }
         }

         /* Note that we don't care if the three units we manage for each user object are up or not, as we are
