Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: test: drop path_is_network_fs_harder
Forwarded: https://github.com/systemd/systemd/pull/35065
--- a/src/test/test-mount-util.c
+++ b/src/test/test-mount-util.c
@@ -537,10 +537,4 @@ TEST(bind_mount_submounts) {
         assert_se(umount_recursive(b, 0) >= 0);
 }
 
-TEST(path_is_network_fs_harder) {
-        ASSERT_OK_ZERO(path_is_network_fs_harder("/dev"));
-        ASSERT_OK_ZERO(path_is_network_fs_harder("/sys"));
-        ASSERT_OK_ZERO(path_is_network_fs_harder("/run"));
-}
-
 DEFINE_TEST_MAIN(LOG_DEBUG);
