Author: Nick Rosbrook <enr0n@ubuntu.com>
Description: test: set nsec3-salt-length=8 in knot.conf
Origin: upstream, commit:59e5108fb4e61957cb40bb15ac7966d085d13af2)
Forwarded: yes
--- a/test/knot-data/knot.conf
+++ b/test/knot-data/knot.conf
@@ -60,6 +60,7 @@
       ksk-submission: parent_zone_sbm
       nsec3-iterations: 0
       nsec3: on
+      nsec3-salt-length: 8
       propagation-delay: 1s
       signing-threads: 4
       zone-max-ttl: 1s
