Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: test-audit-util: skip failing assertion in chroot
Forwarded: https://github.com/systemd/systemd/pull/35188
--- a/src/test/test-audit-util.c
+++ b/src/test/test-audit-util.c
@@ -2,6 +2,7 @@
 
 #include "audit-util.h"
 #include "tests.h"
+#include "virt.h"
 
 TEST(audit_loginuid_from_pid) {
         _cleanup_(pidref_done) PidRef self = PIDREF_NULL, pid1 = PIDREF_NULL;
@@ -24,6 +25,9 @@
         if (r >= 0)
                 log_info("self audit session id: %" PRIu32, sessionid);
 
+        if (running_in_chroot() > 0)
+                return; /* Debian package build in chroot, next assert will fail, skip */
+
         assert_se(audit_session_from_pid(&pid1, &sessionid) == -ENODATA);
 }
 
